packages/components/src/internal/components/assay/AssayWizardModel.ts

```diff
@@ -239,4 +235,78 @@ export class AssayWizardModel
 
         return assayData;
     }
+
+    getInitialQueryModelData(): { rows: { [key: string]: any }; orderedRows: string[] } {
+        const { assayDef, selectedSamples } = this;
+        const sampleColumnData = assayDef.getSampleColumn();
+        const sampleColInResults = sampleColumnData && sampleColumnData.domain === AssayDomainTypes.RESULT;
+        const hasSamples = sampleColInResults && selectedSamples;
+        // We only care about passing samples to the data grid if there is a sample column in the results domain.
+        const rows = hasSamples ? selectedSamples.toJS() : {};
+        return { rows, orderedRows: Object.keys(rows) };
+    }
+
+    // TODO: this method is essentially a duplicate of loadDataForEditor. We should probably make a version that can be
```

Comment:
Yes, probably a later story, but I may just factor out the duplicate bits as part of this PR. The only reason I didn't rush to do it was I wasn't sure how similar it would remain at the time.